vgen/src/main/java/io/gapi/vgen/GapicLanguageProvider.java

```diff
@@ -26,24 +24,19 @@
  * A GapicLanguageProvider performs code or fragment generation using on a proto-based Model for a
  * particular language.
  */
-public interface GapicLanguageProvider {
+public interface GapicLanguageProvider<T extends ProtoElement> {
 
   Model getModel();
 
   /**
-   * Generates the code for the given interface using the given snippet.
+   * Gets the viewing of the model.
    */
-  GeneratedResult generateCode(Interface iface, SnippetDescriptor snippetDescriptor);
+  public ProtoElementViewing<T> getViewing();
 
   /**
-   * Generates the doc for the given proto file and snippet.
+   * Runs code generation.
    */
-  GeneratedResult generateDoc(ProtoFile file, SnippetDescriptor descriptor);
-
-  /**
-   * Generates a fragment for the given method.
-   */
-  GeneratedResult generateFragments(Method method, SnippetDescriptor snippetDescriptor);
+  GeneratedResult generate(T t, SnippetDescriptor snippetDescriptor);
```

Comment:
After reading more of the code, it looks like the type might best be 'InputElementT' and the variable might best be 'element' or 'inputElement'.